boot/boot.go

```diff
@@ -360,5 +358,44 @@ func MakeBootable(model *asserts.Model, rootdir string, bootWith *BootableSet) e
 	}
 
 	return nil
+}
+
+func makeBootable20(model *asserts.Model, rootdir string, bootWith *BootableSet, opts *bootloader.Options) error {
+	// we can only make a single recovery system bootable right now
+	recoverySystems, err := filepath.Glob(filepath.Join(rootdir, "systems/*"))
+	if err != nil {
+		return fmt.Errorf("cannot validate recovery systems: %v", err)
+	}
+	if len(recoverySystems) > 1 {
+		return fmt.Errorf("cannot make multiple recovery systems bootable yet")
+	}
+
+	// install the bootloader configuration from the gadget
+	if err := bootloader.InstallBootConfig(bootWith.UnpackedGadgetDir, rootdir, opts); err != nil {
+		return err
+	}
+
+	// XXX: extract kernel for e.g. ARM
+	//
+	// XXX2: "pseudo" symlink kernel to /systems/XXXX/kernel and add
+	//       code to grub to read it?
+
+	return nil
+}
+
+// MakeBootable sets up the image filesystem with the given rootdir
+// such that it can be booted.
+func MakeBootable(model *asserts.Model, rootdir string, bootWith *BootableSet) error {
+	opts := &bootloader.Options{
+		// XXX: allow to override this
+		PrepareImageTime: true,
+	}
+
+	if model.Grade() == asserts.ModelGradeUnset {
```

Comment:
I think this warrants a comment, having model grade of any value explicitly opts into the core20 ecosystem.